docker_cli/_image.py

```diff
@@ -0,0 +1,471 @@
+import io
+import os
+from shlex import split
+from subprocess import DEVNULL, PIPE, CalledProcessError, run
+from sys import stdin
+from typing import (Any, Iterable, List, Optional, Type, TypeVar, Union,
+                    overload)
+
+from ._exceptions import (CommandNotFoundError, DockerBuildError,
+                          ImageNotFoundError)
+
+
+class Image:
+    """
+    A docker image.
+
+    Holds a reference to a given docker image and provides
+    an interface by which to interact with that image.
+
+    Capabilities include:
+    -   Building docker images from dockerfiles or dockerfile-formatted strings
+        via :func:`~docker_cli.Image.build`.
+    -   Running commands in containers built from the image using
+        :func:`~docker_cli.Image.run`.
+    -   Inspecting properties of the given image.
+    """
+    Self = TypeVar("Self", bound="Image")
+
+    def __init__(self, name_or_id: str):
+        """
+        Initialize a new Image object.
+
+        Parameters
+        ----------
+        name_or_id : str
+            A name or ID by which to find this image using docker inspect.
+        """
+        self._id = get_image_id(name_or_id)
+
+    @overload
+    @classmethod
+    def build(
+        cls: Type[Self],
+        tag: str,
+        *,
+        context: Union[str, os.PathLike[str]],
+        dockerfile: Optional[os.PathLike[str]],
+        stdout: Optional[io.TextIOBase],
+        stderr: Optional[io.TextIOBase],
+        network: str,
+        no_cache: bool
+    ) -> Self:
+        """
+        Build a new image from a dockerfile.
+
+        Build a Dockerfile at the given path with the given name, then
+        return the associated Image instance.
+
+        Parameters
+        ----------
+        tag : str
+            A name for the image.
+        context : os.PathLike, optional
+            The build context. Defaults to ".".
+        dockerfile : os.PathLike, optional
+            The path of the Dockerfile to build, relative to the `context`
+            directory. Defaults to None.
```

Comment:
`dockerfile` is required in this overload.

In general, exactly one of `dockerfile` or `dockerfile_string` must be supplied, but this docstring is just describing the `dockerfile` case.

```suggestion
        dockerfile : os.PathLike
            The path of the Dockerfile to build, relative to the `context`
            directory.
```